aioxmpp/service.py

```diff
@@ -263,14 +258,73 @@ def __get__(self, instance, owner):
         return obj
 
 
+class DependencyGraphNode:
+    def __init__(self):
+        self.class_ = None
+
+    def __repr__(self):
+        return "DependencyGraphNode({!r})".format(self.class_)
```

Comment:
Do I understand it correctly that you introduced this indirection to deal with classes which are currently being constructed (and not constructed yet)? Otherwise I’d suggest to remove this class.